frontend/src/buttons/LaunchButton/LaunchButton.tsx

```diff
@@ -25,6 +25,14 @@ export const LaunchButton: React.FC<Props> = ({ connection, service }) => {
   const [open, setOpen] = useState<boolean>(false)
   const [username, setUsername] = useState<string>((connection && connection.username) || '')
 
+  useEffect(() => {
+    if(connection && connection.username) {
+      setUsername(connection.username)
+    } else {
+      setUsername('');
```

Comment:
could you avoid `;` ?

Reply:
I think you could use the same condition that inline 26 `(connection && connection.username) || ''`